### PR TITLE
[react-vega] Remove, bundles its own types

### PR DESCRIPTION
**Pull request by BendingBender**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: notNeededPackages.json [hunk 1]
```diff
@@ -3598,6 +3598,10 @@
             "libraryName": "react-tooltip",
             "asOfVersion": "4.2.4"
         },
+        "react-vega": {
+            "libraryName": "react-vega",
+            "asOfVersion": "7.0.0"
```
Comment (bmatcuk) [inline notNeededPackages.json:3603]: According to the package.json, types were not included until v7.4.2: https://github.com/vega/react-vega/blob/v7.4.2/packages/react-vega/package.json#L14

**Review by bmatcuk — CHANGES_REQUESTED** [review]

File: notNeededPackages.json (hunk 1 above)
Comment (BendingBender) [reply notNeededPackages.json:3603]: please check via `npm`, the tags in the repos are not always accurate: `npm pack react-vega@7.0.0`

Comment (bmatcuk) [reply notNeededPackages.json:3603]: Oh, I see... I thought typescript required the "types" property, but it'll pick them up if the file is named correctly, too. My bad.

**Review by bmatcuk — APPROVED** [review]
lgtm

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by BendingBender** [comment]
Ready to merge

**Pull request merged**